### PR TITLE
Add a recipe for nanopack.

### PR DESCRIPTION
**Pull request by martin-g**
A meta package for several other recipes by the same author

Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/nanopack/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+requirements:
+  run:
+    - nanoplot
+    - nanostat
+    - nanofilt
+    - nanolyse
+    - nanoget
+    - nanomath
+    - nanocomp
+    - nanogc
```
Comment (martin-g) [inline recipes/nanopack/meta.yaml:27]:
```suggestion
    - nanoqc
```

File: recipes/nanopack/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  version: {{ version }}
+
+build:
+  number: 0
+  noarch: generic
+  run_exports:
+    - {{ pin_subpackage(name, max_pin='x') }}
+
+source:
+  git_url: https://github.com/kehrlab/PopIns2
```
Comment (martin-g) [inline recipes/nanopack/meta.yaml:15]:
```suggestion
  git_url: https://github.com/wdecoster/nanopack
```

**Comment by martin-g** [comment]
@BiocondaBot please fetch artifacts

**Comment by martin-g** [comment]
```
nanopack/packages/noarch took 32s 
❯ ll
.rw-r--r-- martin martin 556 B  Mon Dec 23 11:10:08 2024  current_repodata.json
.rw-r--r-- martin martin 382 B  Mon Dec 23 11:10:08 2024  current_repodata.json.bz2
.rw-r--r-- martin martin 2.5 KB Mon Dec 23 11:10:08 2024  index.html
drwxrwxr-x martin martin 4.0 KB Mon Dec 23 14:19:16 2024  info
.rw-r--r-- martin martin  18 KB Mon Dec 23 11:10:08 2024  nanopack-1.1.1-hdfd78af_0.tar.bz2
.rw-r--r-- martin martin 556 B  Mon Dec 23 11:10:08 2024  repodata.json
.rw-r--r-- martin martin 382 B  Mon Dec 23 11:10:08 2024  repodata.json.bz2
.rw-r--r-- martin martin 556 B  Mon Dec 23 11:10:08 2024  repodata_from_packages.json
.rw-r--r-- martin martin 382 B  Mon Dec 23 11:10:08 2024  repodata_from_packages.json.bz2

nanopack/packages/noarch 
❯ conda install -c ./ nanopack
Channels:
 - file:///tmp/nanopack/packages
 - bioconda
 - conda-forge
Platform: linux-64
Collecting package metadata (repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/martin/miniforge3

  added / updated specs:
    - nanopack


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    aws-c-auth-0.8.0           |      h8c8080f_14         105 KB  conda-forge
    aws-c-cal-0.8.1            |       h0f28dba_2          47 KB  conda-forge
    aws-c-common-0.10.5        |       hb9d3cd8_0         232 KB  conda-forge
    aws-c-compression-0.3.0    |       h9cc6398_4          19 KB  conda-forge
    aws-c-event-stream-0.5.0   |      hf811eff_10          53 KB  conda-forge
    aws-c-http-0.9.2           |       hce7dc5d_3         193 KB  conda-forge
    aws-c-io-0.15.3            |       hfd54f12_3         154 KB  conda-forge
    aws-c-mqtt-0.11.0          |      ha3c2ba9_11         189 KB  conda-forge
    aws-c-s3-0.7.5             |       h55e9418_4         111 KB  conda-forge
    aws-c-sdkutils-0.2.1       |       h9cc6398_3          55 KB  conda-forge
    aws-checksums-0.2.2        |       h9cc6398_3          71 KB  conda-forge
    aws-crt-cpp-0.29.7         |       hed26007_5         346 KB  conda-forge
    aws-sdk-cpp-1.11.458       |       h571fd1c_3         2.9 MB  conda-forge
    azure-core-cpp-1.14.0      |       h5cfcd09_0         337 KB  conda-forge
    azure-identity-cpp-1.10.0  |       h113e628_0         227 KB  conda-forge
    azure-storage-blobs-cpp-12.13.0|       h3cf044e_1         536 KB  conda-forge
    azure-storage-common-cpp-12.8.0|       h736e048_1         146 KB  conda-forge
    azure-storage-files-datalake-cpp-12.12.0|       ha633028_1         281 KB  conda-forge
    biopython-1.84             |  py312h9a8786e_0         3.3 MB  conda-forge
    ca-certificates-2024.12.14 |       hbcca054_0         153 KB  conda-forge
    certifi-2024.12.14         |     pyhd8ed1ab_0         158 KB  conda-forge
    conda-24.11.2              |  py312h7900ff3_0         1.1 MB  conda-forge
    cramino-0.15.0             |       h3dc2dae_1         4.1 MB  bioconda
    deprecated-1.2.15          |     pyhd8ed1ab_1          14 KB  conda-forge
    expat-2.6.4                |       h5888daf_0         135 KB  conda-forge
    font-ttf-dejavu-sans-mono-2.37|       hab24e00_0         388 KB  conda-forge
    font-ttf-inconsolata-3.000 |       h77eed37_0          94 KB  conda-forge
    font-ttf-source-code-pro-2.038|       h77eed37_0         684 KB  conda-forge
    font-ttf-ubuntu-0.83       |       h77eed37_3         1.5 MB  conda-forge
    fontconfig-2.15.0          |       h7e30c49_1         259 KB  conda-forge
    fonts-conda-forge-1        |                0           4 KB  conda-forge
    gflags-2.2.2               |    h5888daf_1005         117 KB  conda-forge
    glog-0.7.1                 |       hbabe93e_0         140 KB  conda-forge
    kaleido-core-0.2.1         |       h3644ca4_0        59.2 MB  conda-forge
    libabseil-20240722.0       | cxx17_h5888daf_1         1.2 MB  conda-forge
    libarrow-18.1.0            |   h3b07799_4_cpu         8.4 MB  conda-forge
    libarrow-acero-18.1.0      |   h8bbc2ab_4_cpu         596 KB  conda-forge
    libarrow-dataset-18.1.0    |   h8bbc2ab_4_cpu         572 KB  conda-forge
    libarrow-substrait-18.1.0  |   had74209_4_cpu         508 KB  conda-forge
    libblas-3.9.0              |26_linux64_openblas          16 KB  conda-forge
    libcblas-3.9.0             |26_linux64_openblas          16 KB  conda-forge
    libcrc32c-1.1.2            |       h9c3ff4c_0          20 KB  conda-forge
    libdeflate-1.23            |       h4ddbbb0_0          71 KB  conda-forge
    libevent-2.1.12            |       hf998b51_1         417 KB  conda-forge
    libgoogle-cloud-2.31.0     |       h804f50b_0         1.2 MB  conda-forge
    libgoogle-cloud-storage-2.31.0|       h0121fbd_0         764 KB  conda-forge
    libgrpc-1.67.1             |       hc2c308b_0         7.0 MB  conda-forge
    liblapack-3.9.0            |26_linux64_openblas          16 KB  conda-forge
    libparquet-18.1.0          |   hf4f6db6_4_cpu         1.1 MB  conda-forge
    libprotobuf-5.28.2         |       h5b01275_0         2.8 MB  conda-forge
    libre2-11-2024.07.02       |       hbbce691_1         206 KB  conda-forge
    libthrift-0.21.0           |       h0e7cc3e_0         416 KB  conda-forge
    libutf8proc-2.9.0          |       hb9d3cd8_1          80 KB  conda-forge
    mappy-2.28                 |  py312h4711d71_3         171 KB  bioconda
    mathjax-2.7.7              |       ha770c72_3        21.2 MB  conda-forge
    nanocomp-1.24.2            |     pyhdfd78af_0          24 KB  bioconda
    nanofilt-2.8.0             |             py_0          12 KB  bioconda
    nanoget-1.19.3             |     pyhdfd78af_0          34 KB  bioconda
    nanolyse-1.2.1             |     pyhdfd78af_0          41 KB  bioconda
    nanomath-1.4.0             |     pyhdfd78af_0          29 KB  bioconda
    nanopack-1.1.1             |       hdfd78af_0          18 KB  file:///tmp/nanopack/packages
    nanoplot-1.43.0            |     pyhdfd78af_1          37 KB  bioconda
    nanoqc-0.10.0              |     pyhdfd78af_0          27 KB  bioconda
    nanostat-1.6.0             |     pyhdfd78af_0          12 KB  bioconda
    nspr-4.36                  |       h5888daf_0         225 KB  conda-forge
    nss-3.107                  |       hdf54f9c_0         1.9 MB  conda-forge
    numpy-2.2.1                |  py312h7e784f5_0         8.1 MB  conda-forge
    orc-2.0.3                  |       he039a57_0         1.1 MB  conda-forge
    pandas-2.2.3               |  py312hf9745cd_1        14.7 MB  conda-forge
    plotly-5.24.1              |     pyhd8ed1ab_1         7.7 MB  conda-forge
    psutil-6.1.0               |  py312h66e93f0_0         477 KB  conda-forge
    pyarrow-18.1.0             |  py312h7900ff3_0          25 KB  conda-forge
    pyarrow-core-18.1.0        |py312h01725c0_0_cpu         4.4 MB  conda-forge
    pysam-0.22.1               |  py312hcfdcdd7_2         4.4 MB  bioconda
    python-dateutil-2.9.0.post0|     pyhff2d567_1         217 KB  conda-forge
    python-kaleido-0.2.1       |     pyhd8ed1ab_0          18 KB  conda-forge
    python-tzdata-2024.2       |     pyhd8ed1ab_1         139 KB  conda-forge
    pytz-2024.1                |     pyhd8ed1ab_0         184 KB  conda-forge
    re2-2024.07.02             |       h77b4e00_1          26 KB  conda-forge
    s2n-1.5.9                  |       h0fd0ee4_0         347 KB  conda-forge
    scipy-1.14.1               |  py312h62794b6_2        16.6 MB  conda-forge
    six-1.17.0                 |     pyhd8ed1ab_0          16 KB  conda-forge
    snappy-1.2.1               |       h8bd8927_1          42 KB  conda-forge
    sqlite-3.47.0              |       h9eae976_1         863 KB  conda-forge
    tenacity-9.0.0             |     pyhd8ed1ab_1          24 KB  conda-forge
    wrapt-1.17.0               |  py312h66e93f0_0          62 KB  conda-forge
    ------------------------------------------------------------
                                           Total:       185.9 MB

The following NEW packages will be INSTALLED:

  aws-c-auth         conda-forge/linux-64::aws-c-auth-0.8.0-h8c8080f_14 
  aws-c-cal          conda-forge/linux-64::aws-c-cal-0.8.1-h0f28dba_2 
  aws-c-common       conda-forge/linux-64::aws-c-common-0.10.5-hb9d3cd8_0 
  aws-c-compression  conda-forge/linux-64::aws-c-compression-0.3.0-h9cc6398_4 
  aws-c-event-stream conda-forge/linux-64::aws-c-event-stream-0.5.0-hf811eff_10 
  aws-c-http         conda-forge/linux-64::aws-c-http-0.9.2-hce7dc5d_3 
  aws-c-io           conda-forge/linux-64::aws-c-io-0.15.3-hfd54f12_3 
  aws-c-mqtt         conda-forge/linux-64::aws-c-mqtt-0.11.0-ha3c2ba9_11 
  aws-c-s3           conda-forge/linux-64::aws-c-s3-0.7.5-h55e9418_4 
  aws-c-sdkutils     conda-forge/linux-64::aws-c-sdkutils-0.2.1-h9cc6398_3 
  aws-checksums      conda-forge/linux-64::aws-checksums-0.2.2-h9cc6398_3 
  aws-crt-cpp        conda-forge/linux-64::aws-crt-cpp-0.29.7-hed26007_5 
  aws-sdk-cpp        conda-forge/linux-64::aws-sdk-cpp-1.11.458-h571fd1c_3 
  azure-core-cpp     conda-forge/linux-64::azure-core-cpp-1.14.0-h5cfcd09_0 
  azure-identity-cpp conda-forge/linux-64::azure-identity-cpp-1.10.0-h113e628_0 
  azure-storage-blo~ conda-forge/linux-64::azure-storage-blobs-cpp-12.13.0-h3cf044e_1 
  azure-storage-com~ conda-forge/linux-64::azure-storage-common-cpp-12.8.0-h736e048_1 
  azure-storage-fil~ conda-forge/linux-64::azure-storage-files-datalake-cpp-12.12.0-ha633028_1 
  biopython          conda-forge/linux-64::biopython-1.84-py312h9a8786e_0 
  cramino            bioconda/linux-64::cramino-0.15.0-h3dc2dae_1 
  deprecated         conda-forge/noarch::deprecated-1.2.15-pyhd8ed1ab_1 
  expat              conda-forge/linux-64::expat-2.6.4-h5888daf_0 
  font-ttf-dejavu-s~ conda-forge/noarch::font-ttf-dejavu-sans-mono-2.37-hab24e00_0 
  font-ttf-inconsol~ conda-forge/noarch::font-ttf-inconsolata-3.000-h77eed37_0 
  font-ttf-source-c~ conda-forge/noarch::font-ttf-source-code-pro-2.038-h77eed37_0 
  font-ttf-ubuntu    conda-forge/noarch::font-ttf-ubuntu-0.83-h77eed37_3 
  fontconfig         conda-forge/linux-64::fontconfig-2.15.0-h7e30c49_1 
  fonts-conda-forge  conda-forge/noarch::fonts-conda-forge-1-0 
  freetype           conda-forge/linux-64::freetype-2.12.1-h267a509_2 
  gflags             conda-forge/linux-64::gflags-2.2.2-h5888daf_1005 
  glog               conda-forge/linux-64::glog-0.7.1-hbabe93e_0 
  kaleido-core       conda-forge/linux-64::kaleido-core-0.2.1-h3644ca4_0 
  libabseil          conda-forge/linux-64::libabseil-20240722.0-cxx17_h5888daf_1 
  libarrow           conda-forge/linux-64::libarrow-18.1.0-h3b07799_4_cpu 
  libarrow-acero     conda-forge/linux-64::libarrow-acero-18.1.0-h8bbc2ab_4_cpu 
  libarrow-dataset   conda-forge/linux-64::libarrow-dataset-18.1.0-h8bbc2ab_4_cpu 
  libarrow-substrait conda-forge/linux-64::libarrow-substrait-18.1.0-had74209_4_cpu 
  libblas            conda-forge/linux-64::libblas-3.9.0-26_linux64_openblas 
  libbrotlicommon    conda-forge/linux-64::libbrotlicommon-1.1.0-hb9d3cd8_2 
  libbrotlidec       conda-forge/linux-64::libbrotlidec-1.1.0-hb9d3cd8_2 
  libbrotlienc       conda-forge/linux-64::libbrotlienc-1.1.0-hb9d3cd8_2 
  libcblas           conda-forge/linux-64::libcblas-3.9.0-26_linux64_openblas 
  libcrc32c          conda-forge/linux-64::libcrc32c-1.1.2-h9c3ff4c_0 
  libdeflate         conda-forge/linux-64::libdeflate-1.23-h4ddbbb0_0 
  libevent           conda-forge/linux-64::libevent-2.1.12-hf998b51_1 
  libgfortran        conda-forge/linux-64::libgfortran-14.2.0-h69a702a_1 
  libgfortran5       conda-forge/linux-64::libgfortran5-14.2.0-hd5240d6_1 
  libgoogle-cloud    conda-forge/linux-64::libgoogle-cloud-2.31.0-h804f50b_0 
  libgoogle-cloud-s~ conda-forge/linux-64::libgoogle-cloud-storage-2.31.0-h0121fbd_0 
  libgrpc            conda-forge/linux-64::libgrpc-1.67.1-hc2c308b_0 
  liblapack          conda-forge/linux-64::liblapack-3.9.0-26_linux64_openblas 
  libopenblas        conda-forge/linux-64::libopenblas-0.3.28-pthreads_h94d23a6_1 
  libparquet         conda-forge/linux-64::libparquet-18.1.0-hf4f6db6_4_cpu 
  libpng             conda-forge/linux-64::libpng-1.6.44-hadc24fc_0 
  libprotobuf        conda-forge/linux-64::libprotobuf-5.28.2-h5b01275_0 
  libre2-11          conda-forge/linux-64::libre2-11-2024.07.02-hbbce691_1 
  libthrift          conda-forge/linux-64::libthrift-0.21.0-h0e7cc3e_0 
  libutf8proc        conda-forge/linux-64::libutf8proc-2.9.0-hb9d3cd8_1 
  mappy              bioconda/linux-64::mappy-2.28-py312h4711d71_3 
  mathjax            conda-forge/linux-64::mathjax-2.7.7-ha770c72_3 
  nanocomp           bioconda/noarch::nanocomp-1.24.2-pyhdfd78af_0 
  nanofilt           bioconda/noarch::nanofilt-2.8.0-py_0 
  nanoget            bioconda/noarch::nanoget-1.19.3-pyhdfd78af_0 
  nanolyse           bioconda/noarch::nanolyse-1.2.1-pyhdfd78af_0 
  nanomath           bioconda/noarch::nanomath-1.4.0-pyhdfd78af_0 
  nanopack           packages/noarch::nanopack-1.1.1-hdfd78af_0 
  nanoplot           bioconda/noarch::nanoplot-1.43.0-pyhdfd78af_1 
  nanoqc             bioconda/noarch::nanoqc-0.10.0-pyhdfd78af_0 
  nanostat           bioconda/noarch::nanostat-1.6.0-pyhdfd78af_0 
  nspr               conda-forge/linux-64::nspr-4.36-h5888daf_0 
  nss                conda-forge/linux-64::nss-3.107-hdf54f9c_0 
  numpy              conda-forge/linux-64::numpy-2.2.1-py312h7e784f5_0 
  orc                conda-forge/linux-64::orc-2.0.3-he039a57_0 
  pandas             conda-forge/linux-64::pandas-2.2.3-py312hf9745cd_1 
  plotly             conda-forge/noarch::plotly-5.24.1-pyhd8ed1ab_1 
  psutil             conda-forge/linux-64::psutil-6.1.0-py312h66e93f0_0 
  pyarrow            conda-forge/linux-64::pyarrow-18.1.0-py312h7900ff3_0 
  pyarrow-core       conda-forge/linux-64::pyarrow-core-18.1.0-py312h01725c0_0_cpu 
  pysam              bioconda/linux-64::pysam-0.22.1-py312hcfdcdd7_2 
  python-dateutil    conda-forge/noarch::python-dateutil-2.9.0.post0-pyhff2d567_1 
  python-kaleido     conda-forge/noarch::python-kaleido-0.2.1-pyhd8ed1ab_0 
  python-tzdata      conda-forge/noarch::python-tzdata-2024.2-pyhd8ed1ab_1 
  pytz               conda-forge/noarch::pytz-2024.1-pyhd8ed1ab_0 
  re2                conda-forge/linux-64::re2-2024.07.02-h77b4e00_1 
  s2n                conda-forge/linux-64::s2n-1.5.9-h0fd0ee4_0 
  scipy              conda-forge/linux-64::scipy-1.14.1-py312h62794b6_2 
  six                conda-forge/noarch::six-1.17.0-pyhd8ed1ab_0 
  snappy             conda-forge/linux-64::snappy-1.2.1-h8bd8927_1 
  sqlite             conda-forge/linux-64::sqlite-3.47.0-h9eae976_1 
  tenacity           conda-forge/noarch::tenacity-9.0.0-pyhd8ed1ab_1 
  wrapt              conda-forge/linux-64::wrapt-1.17.0-py312h66e93f0_0 

The following packages will be UPDATED:

  ca-certificates                      2024.8.30-hbcca054_0 --> 2024.12.14-hbcca054_0 
  certifi                            2024.8.30-pyhd8ed1ab_0 --> 2024.12.14-pyhd8ed1ab_0 
  conda                             24.11.0-py312h7900ff3_0 --> 24.11.2-py312h7900ff3_0 


Proceed ([y]/n)? y


Downloading and Extracting Packages:
                                                                                                                                                                                                                     
Preparing transaction: done                                                                                                                                                                                          
Verifying transaction: done                                                                                                                                                                                          
Executing transaction: done                                                                                                                                                                                          
```
LGTM!

**Comment by martin-g** [comment]
@BiocondaBot please add label

**Comment by bgruening** [comment]
What is the advantage of this package in contrast to a environment.yaml file? Why do we need to clone the github repo, if its just a meta-package?

**Comment by martin-g** [comment]
> What is the advantage of this package in contrast to a environment.yaml file?

By using a recipe it is created/maintained at one place. Using a env file will have to be created by each user and maintained by them.

There are several other recipes like this:
```
./dipcall/meta.yaml
./atlas-gene-annotation-manipulation/meta.yaml
./lepwrap/meta.yaml
./haplogrep/meta.yaml
./pstrain/meta.yaml
./atlas-fastq-provider/meta.yaml
```



> Why do we need to clone the github repo, if its just a meta-package?

Hm, I haven't tried without `source:` but I'd expect that it will fail. I use `git_rev` because there is no release/tag in that repo.

**Review by bgruening — APPROVED** [review]
Ok :)

**Pull request merged**